src/main/kotlin/com/example/delivery/domain/DomainOrder.kt

```diff
@@ -0,0 +1,14 @@
+package com.example.delivery.domain
+
+import com.example.delivery.mongo.MongoOrder.MongoShipmentDetails
+import org.bson.types.ObjectId
+import java.math.BigDecimal
+
+data class DomainOrder(
+    val id: ObjectId? = null,
```

Comment:
can your domain model exist without an id? 
```suggestion
    val id: ObjectId,
```